docs/source/hyper-parameter-search.rst

```diff
@@ -1,33 +1,193 @@
+2. "**compute constrained**", which happen when the computation takes too long
+   even with data that can fit in memory.  This typically happens when many
+   hyperparameters need to be tuned, which especially common with neural
+   networks.
+
+Here's an example of what "memory constrained" means:
+
+.. code:: python
+
+   # not memory constrained
+   import pandas as pd
+   df = pd.read_csv("0.parquet")
+   print(df.shape)  # (30000, 200) => 23MB
+
+   # memory constrained
+   import dask.dataframe as dd
+   # Read 1000 of the above dataframes (22GB of data)
+   ddf = dd.read_parquet("*.parquet")
+
+Here's an example of what "compute constrained" means. Let's look at when
+
+.. code:: python
+
+   from scipy.stats import uniform, loguniform
+   from sklearn.linear_model import SGDClasifier
+
+   model = SGDClasifier()
+
+   # not compute constrained
+   params = {"l1_ratio": uniform(0, 1)}
+
+   # compute constrained
+   params = {
+       "l1_ratio": uniform(0, 1),
+       "alpha": loguniform(1e-5, 1e-1),
+       "penalty": ["l2", "l1", "elasticnet"],
+       "learning_rate": ["invscaling", "adaptive"],
+       "power_t": uniform(0, 1),
+       "average": [True, False],
+   }
+
+These issues are independent and both can happen the same time. Dask-ML has
+tools to address all 4 combinations. Let's look at each case.
+
+Neither compute nor memory constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when there aren't many hyperparameters to tune and the data
+fits in memory. This is common when the search can run on your laptop.
+
+Scikit-learn can handle this case:
+
+.. autosummary::
+   sklearn.model_selection.GridSearchCV
+   sklearn.model_selection.RandomizedSearchCV
+
+Dask-ML also has some drop in replacements for the Scikit-learn versions that
+works well with Dask.
+
+.. autosummary::
+   dask_ml.model_selection.GridSearchCV
+   dask_ml.model_selection.RandomizedSearchCV
+
+
+These models work well with Dask Arrays/DataFrames. By default, these
+estimators will pass the entire dataset to ``fit`` even if a Dask
+Array/DataFrame is passed.  More detail is in
+":ref:`works-with-dask-collections`".
+
+These estimators make some progress into hyperparameter searches that are
+"compute constrained" or "memory constrained". Details on the cases are
+mentioned below in ":ref:`hyperparameter.mem-ncpu`"and
+":ref:`hyperparameter.cpu-nmem`".
+
+.. _hyperparameter.mem-ncpu:
+
+Memory constrained, but not compute constrained
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+This case happens when the data doesn't fit in memory but there aren't many
+hyperparameters to search over. The data doesn't fit in memory, so it makes
+sense to call ``partial_fit`` on each chunk of a Dask Array/Dataframe. These
+estimators do that, possibly with some configuration:
+
+.. autosummary::
+   dask_ml.model_selection.IncrementalSearchCV
+   dask_ml.model_selection.RandomizedSearchCV
+   dask_ml.model_selection.GridSearchCV
```

Comment:
I'm not sure I would put these here. The caveat about Incremental is a big one. I think these fall more naturally under "Compute, but not memory constrained".

Reply:
Thanks. Let me know if you have another edits on where different estimators should go.